index.ts

```diff
@@ -386,33 +372,22 @@ class Analytics {
         Authorization: 'Basic ' + Buffer.from(`${this.writeKey}:`).toString('base64'),
       },
       body: JSON.stringify(data),
+      timeout,
       retryDelay: this._exponentialDelay.bind(this),
       retryOn: this._isErrorRetryable.bind(this),
     };
 
-    if (this.timeout) {
-      // @ts-expect-error Need to define timeout on the request object
-      req.timeout = typeof this.timeout === 'string' ? ms(this.timeout) : this.timeout;
-    }
-
     retryableFetch(`${this.host}`, req)
-      .then((_response) => {
-        this.queue.splice(0, items.length);
```

Comment:
Removing this means that if there's an error delivering events, they'll get lost forever instead of eventually retrying, and later events may succeed. Having holes in a sequence of events seems like it could be an issue.

Reply:
Just speaking from my opinion - the retry logic is already a part of the fetch request. So,if we get to the point that a request failed all 4 attempts over 2-3 seconds, I think we can consider it unlikely to succeed if we try again and just ditch those events. Kinda in favor of the simpler code in this case.

Reply:
seems the only time events are not spliced is if the persistence queue throws an error. since we've removed the persistence queue, i think it's fine to just splice beforehand.